### PR TITLE
carta HDF5Image

### PR DESCRIPTION
**Pull request by pford**
Issue #175 - casacore::HDF5Image is incompatible with IDIA's HDF5 schema.  The new class carta::HDF5Image is derived from casacore::ImageInterface.  Works with current casacore::Lattice-based implementation.

Swizzled data in HDF5Loader remains an HDF5Lattice, coordinate system not needed.  Some hdf5 attributes-related code was moved from FileInfoLoader to HDF5Attributes.  Also fixed the cursor z-profile message with hdf5 swizzled data from the loader.

### CODE REVIEW

**Comment by veggiesaurus** [comment]
Wonder if we should call it something less vague than `carta::HDF5Image` to avoid potential clashes or confusion, despite them existing in separate namespaces. `IDIAHDF5Image`? (that's not a great name either, not sure what other options are)

**Comment by confluence** [comment]
How does this image class differ from `casacore::HDF5Image`? Is it only a matter of changing the hardcoded paths to datasets? Could we submit a patch to casacore to make their class more flexible (e.g. by adding optional function parameters to override their defaults)? That doesn't mean we shouldn't proceed with this change in the meantime, though.

**Comment by veggiesaurus** [comment]
@confluence I think the attributes for the cassacore image are stored very differently, IIRC

File: ImageData/HDF5Image.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
+//# HDF5Image.h : CARTA HDF5Image derived from casacore::HDF5Image
```
Comment (confluence) [inline ImageData/HDF5Image.h]: Should this not be `casacore::ImageInterface`?

Comment (pford) [reply ImageData/HDF5Image.h]: Yes, change of plans.  Will fix.

**Comment by pford** [comment]
Looking at the casacore code, it expects a casacore::HDF5Image to be created in a certain way with a casacore::Record containing CoordinateSystem info, ImageInfo, and ObsInfo.  So opening a casacore::HDF5Image means reading in this Record and supplying the subrecord to the appropriate objects.  The name of the data attribute is only one small part.

**Comment by confluence** [comment]
I see now that this looks much simpler than `casacore::HDF5Image`, and also uses a lot of our other custom code. Given how specific this is to our schema, I think this approach makes sense.

**Pull request merged**